File: Demos/Visual Basic .Net/09. Addendas/Adenda/Adenda.vb
Imports System
Imports System.IO
Imports System.Windows.Forms
Imports HyperSoft.ElectronicDocumentLibrary.Axxa.Addenda.Autos
Imports HyperSoft.ElectronicDocumentLibrary.Axxa.Addenda.GastosMedicos
Imports HyperSoft.ElectronicDocumentLibrary.Coppel.Addenda
Imports HyperSoft.ElectronicDocumentLibrary.Document
Imports HyperSoft.ElectronicDocumentLibrary.EnvasesUniversalesMexico.Addenda
Imports HyperSoft.ElectronicDocumentLibrary.Soriana.Remision.Addenda
Imports HyperSoft.ElectronicDocumentLibrary.Soriana.Servicio.Addenda
Imports HyperSoft.Shared


Friend NotInheritable Partial Class Adenda
	#Region "Vars"

	Private Shared directorioSalida As String
	Private Shared electronicDocument As ElectronicDocument

	#End Region

	#Region "Methods"

	Friend Shared Sub SetConfiguration(directorio As String, electronic As ElectronicDocument)
		directorioSalida = directorio
		electronicDocument = electronic
	End Sub

	Private Shared Function Save(fileName As String, ByRef fullFileName As String) As Boolean
		Dim errorMessage As String
		fullFileName = Path.Combine(directorioSalida, fileName)

		If System.IO.File.Exists(fullFileName) Then
			[Shared].File.Instance.DeleteFile(fullFileName, errorMessage)
		End If


		Using stream As New MemoryStream()
			' Se ejecuta el proceso de generación
			If electronicDocument.SaveToStream(stream) = False Then
				errorMessage = String.Format("Se generó un error al crear el XML.{0}{0}ERROR{0}{1}", Environment.NewLine, electronicDocument.ErrorText)
				MessageBox.Show(errorMessage, "Adenda", MessageBoxButtons.OK, MessageBoxIcon.[Error])
				Return False
			End If

			Try
				System.IO.File.WriteAllBytes(fullFileName, stream.ToArray())
				Return True
			Catch e As Exception
				errorMessage = String.Format("Se generó un error al guardar el archivo XML.{0}{0}ARCHIVO{0}{1}{0}{0}ERROR{0}{2}", Environment.NewLine, fullFileName, e.Message)
				MessageBox.Show(errorMessage, "Adenda", MessageBoxButtons.OK, MessageBoxIcon.[Error])
				Return False
			End Try
		End Using
	End Function

	Friend Shared Function GetAddendaVersion(index As Integer) As String
	Using New WaitCursor()
		Select Case index
			Case 0
          Return ElectronicDocumentLibrary.Aba.Addenda.Aba.Version()
			Case 1
				Return ElectronicDocumentLibrary.Ado.Addenda.Ado.Version()
			Case 2
				Return ElectronicDocumentLibrary.AlSuper.Addenda.AlSuper.Version()
			Case 3
				Return ElectronicDocumentLibrary.AltosHornosMexico.Addenda.AltosHornosMexico.Version()
			Case 4
				Return ElectronicDocumentLibrary.Aluprint.Addenda.Aluprint.Version()
			Case 5
				Return ElectronicDocumentLibrary.Amece.Addenda.Amece.Version()
			Case 6
				Return ElectronicDocumentLibrary.Amis.Addenda.Amis.Version()
			Case 7
				Return ElectronicDocumentLibrary.Asofarma.Asonioscoc.Addenda.Asonioscoc.Version()
			Case 8
				Return Autos.Version()
			Case 9
				Return GastosMedicos.Version()
			Case 10
				Return ElectronicDocumentLibrary.Bbva.Addenda.Bbva.Version
			Case 11
				Return ElectronicDocumentLibrary.Bexel.Addenda.Bexel.Version()
			Case 12
				Return ElectronicDocumentLibrary.BrudiFarma.Addenda.BrudiFarma.Version()
			Case 13
				Return ElectronicDocumentLibrary.CapaOzono.Addenda.CapaOzono.Version()
			Case 14
				Return ElectronicDocumentLibrary.Chrysler.Addenda.Chrysler.Version()
			Case 15
				Return ElectronicDocumentLibrary.Comex.Addenda.Comex.Version()
			Case 16
				Return ElectronicDocumentLibrary.ContinentalTire.Addenda.ContinentalTire.Version()
			Case 17
				Return Coppel.Version()
			Case 18
				Return ElectronicDocumentLibrary.CorporateTravelServices.Addenda.CorporateTravelServices.Version()
			Case 19
				Return ElectronicDocumentLibrary.Disney.Addenda.Disney.Version()
			Case 20
				Return ElectronicDocumentLibrary.Elektra.Addenda.Elektra.Version()
			Case 21
				Return ElectronicDocumentLibrary.Emerson.Addenda.Emerson.Version()
			Case 22
				Return ElectronicDocumentLibrary.Emsur.Addenda.EmSur.Version()
			Case 23
				Return EnvasesUniversalesMexico.Version()
			Case 24
				Return ElectronicDocumentLibrary.Femsa.Addenda.Femsa.Version()
			Case 25
				Return ElectronicDocumentLibrary.FerroMexicana.Addenda.FerroMexicana.Version()
			Case 26
				Return ElectronicDocumentLibrary.Fuller.Addenda.Fuller.Version()
			Case 27
				Return ElectronicDocumentLibrary.Gayosso.Addenda.Gayosso.Version()
			Case 28
				Return ElectronicDocumentLibrary.GeneralMotors.Addenda.GeneralMotors.Version()
			Case 29
				Return ElectronicDocumentLibrary.Gomsa.Addenda.Gomsa.Version
			Case 30
				Return ElectronicDocumentLibrary.GrupoModelo.Addenda.GrupoModelo.Version()
			Case 31
				Return ElectronicDocumentLibrary.Heb.Addenda.Heb.Version()
			Case 32
				Return ElectronicDocumentLibrary.Inbursa.Addenda.Inbursa.Version()
			Case 33
				Return ElectronicDocumentLibrary.Itsb.Addenda.ItSmartBusiness.Version()
			Case 34
				Return ElectronicDocumentLibrary.Iusacell.Addenda.Iusacell.Version()
			Case 35
				Return ElectronicDocumentLibrary.Kuehne.Addenda.Kuehne.Version()
			Case 36
				Return ElectronicDocumentLibrary.Lamosa.Addenda.Lamosa.Version()
			Case 37
				Return ElectronicDocumentLibrary.Landsteiner.Addenda.Landsteiner.Version()
			Case 38
				Return ElectronicDocumentLibrary.Loreal.Addenda.Loreal.Version()
			Case 39
				Return ElectronicDocumentLibrary.Lowes.Addenda.Lowes.Version()
			Case 40
				Return ElectronicDocumentLibrary.Mabe.Addenda.Mabe.Version()
			Case 41
				Return ElectronicDocumentLibrary.Maersk.Addenda.Maersk.Version()
			Case 42
				Return ElectronicDocumentLibrary.MultiPack.Addenda.MultiPack.Version()
			Case 43
				Return ElectronicDocumentLibrary.Oxxo.Addenda.Oxxo.Version()
			Case 44
				Return ElectronicDocumentLibrary.Pemex.Addenda.PemexExploracion.Version()
			Case 45
				Return ElectronicDocumentLibrary.Pemex.Addenda.PemexExploracion.Version()
			Case 46
				Return ElectronicDocumentLibrary.Pepsico.Addenda.Pepsico.Version()
			Case 47
				Return ElectronicDocumentLibrary.Pilgrims.Addenda.Pilgrims.Version()
			Case 48
				Return ElectronicDocumentLibrary.Prolec.Addenda.Prolec.Version()
			Case 49
				Return ElectronicDocumentLibrary.Sanmina.Addenda.Sanmina.Version()
			Case 50
				Return ElectronicDocumentLibrary.Santander.Addenda.Santander.Version()
			Case 51
				Return ElectronicDocumentLibrary.SectorPrimario.Addenda.SectorPrimario.Version()
			Case 52
				Return ElectronicDocumentLibrary.SevenEleven.Addenda.SevenEleven.Version()
			Case 53
				Return ElectronicDocumentLibrary.SolerPalau.Addenda.SolerPalau.Version()
			Case 54
				Return CargaRemision.Version()
			Case 55
				Return Servicio.Version()
			Case 56
				Return ElectronicDocumentLibrary.TiendasNeto.Addenda.TiendasNeto.Version()
			Case 57
				Return ElectronicDocumentLibrary.TransportesCastores.Addenda.TransportesCastores.Version()
			Case 58
				Return ElectronicDocumentLibrary.TresM.Addenda.TresM.Version
			Case 59
				Return ElectronicDocumentLibrary.TvAzteca.Addenda.TvAzteca.Version()
			Case 60
				Return ElectronicDocumentLibrary.Viscofan.Addenda.Viscofan.Version
			Case 61
				Return ElectronicDocumentLibrary.Volkswagen.Addenda.Volkswagen.Version()
			Case 62
				Return ElectronicDocumentLibrary.WalMart.Addenda.WalMart.Version
			Case 63
				Return ElectronicDocumentLibrary.ZfMexico.Addenda.ZfMexico.Version()
		End Select

		Return String.Empty
	End Using
End Function

  #End Region
End Class

File: Demos/CSharp/09. Addendas/Adenda/Adenda.cs
using System;
using System.IO;
using System.Windows.Forms;
using HyperSoft.ElectronicDocumentLibrary.Axxa.Addenda.Autos;
using HyperSoft.ElectronicDocumentLibrary.Axxa.Addenda.GastosMedicos;
using HyperSoft.ElectronicDocumentLibrary.Coppel.Addenda;
using HyperSoft.ElectronicDocumentLibrary.Document;
using HyperSoft.ElectronicDocumentLibrary.EnvasesUniversalesMexico.Addenda;
using HyperSoft.ElectronicDocumentLibrary.Soriana.Remision.Addenda;
using HyperSoft.ElectronicDocumentLibrary.Soriana.Servicio.Addenda;
using HyperSoft.Shared;

namespace HyperSoft.Ejemplo.Adendas
{
  internal static partial class Adenda
  {
    #region Vars

    private static string directorioSalida;
    private static ElectronicDocument electronicDocument;

    #endregion

    #region Methods

    internal static void SetConfiguration(string directorio, ElectronicDocument electronic)
    {
      directorioSalida = directorio;
      electronicDocument = electronic;
    }

    private static bool Save(string fileName, out string fullFileName)
    {
      string errorMessage;
      fullFileName = Path.Combine(directorioSalida, fileName);

      if (System.IO.File.Exists(fullFileName))
        Shared.File.Instance.DeleteFile(fullFileName, out errorMessage);


      using (MemoryStream stream = new MemoryStream())
      {
        // Se ejecuta el proceso de generación
        if (electronicDocument.SaveToStream(stream) == false)
        {
          errorMessage = string.Format("Se generó un error al crear el XML.{0}{0}ERROR{0}{1}", Environment.NewLine, electronicDocument.ErrorText);
          MessageBox.Show(errorMessage, "Adenda", MessageBoxButtons.OK, MessageBoxIcon.Error);
          return false;
        }

        try
        {
          System.IO.File.WriteAllBytes(fullFileName, stream.ToArray());
          return true;
        }
        catch (Exception e)
        {
          errorMessage = string.Format("Se generó un error al guardar el archivo XML.{0}{0}ARCHIVO{0}{1}{0}{0}ERROR{0}{2}", Environment.NewLine, fullFileName, e.Message);
          MessageBox.Show(errorMessage, "Adenda", MessageBoxButtons.OK, MessageBoxIcon.Error);
          return false;
        }
      }
    }

    internal static string GetAddendaVersion(int index)
    {
      using (new WaitCursor())
      {
        switch (index)
        {
          case 0: return ElectronicDocumentLibrary.Aba.Addenda.Aba.Version();
          case 1: return ElectronicDocumentLibrary.Ado.Addenda.Ado.Version();
          case 2: return ElectronicDocumentLibrary.AlSuper.Addenda.AlSuper.Version();
          case 3: return ElectronicDocumentLibrary.AltosHornosMexico.Addenda.AltosHornosMexico.Version();
          case 4: return ElectronicDocumentLibrary.Aluprint.Addenda.Aluprint.Version();
          case 5: return ElectronicDocumentLibrary.Amece.Addenda.Amece.Version();
          case 6: return ElectronicDocumentLibrary.Amis.Addenda.Amis.Version();
          case 7: return ElectronicDocumentLibrary.Asofarma.Asonioscoc.Addenda.Asonioscoc.Version();
          case 8: return Autos.Version();
          case 9: return GastosMedicos.Version();
          case 10: return ElectronicDocumentLibrary.Bbva.Addenda.Bbva.Version;
          case 11: return ElectronicDocumentLibrary.Bexel.Addenda.Bexel.Version();
          case 12: return ElectronicDocumentLibrary.BrudiFarma.Addenda.BrudiFarma.Version();          
          case 13: return ElectronicDocumentLibrary.CapaOzono.Addenda.CapaOzono.Version();
          case 14: return ElectronicDocumentLibrary.Chrysler.Addenda.Chrysler.Version();
          case 15: return ElectronicDocumentLibrary.Comex.Addenda.Comex.Version();
          case 16: return ElectronicDocumentLibrary.ContinentalTire.Addenda.ContinentalTire.Version();
          case 17: return Coppel.Version();
          case 18: return ElectronicDocumentLibrary.CorporateTravelServices.Addenda.CorporateTravelServices.Version();
          case 19: return ElectronicDocumentLibrary.Disney.Addenda.Disney.Version();
          case 20: return ElectronicDocumentLibrary.Elektra.Addenda.Elektra.Version();
          case 21: return ElectronicDocumentLibrary.Emerson.Addenda.Emerson.Version();
          case 22: return ElectronicDocumentLibrary.Emsur.Addenda.EmSur.Version();
          case 23: return EnvasesUniversalesMexico.Version();
          case 24: return ElectronicDocumentLibrary.Femsa.Addenda.Femsa.Version();
          case 25: return ElectronicDocumentLibrary.FerroMexicana.Addenda.FerroMexicana.Version();
          case 26: return ElectronicDocumentLibrary.Fuller.Addenda.Fuller.Version();
          case 27: return ElectronicDocumentLibrary.Gayosso.Addenda.Gayosso.Version();
          case 28: return ElectronicDocumentLibrary.GeneralMotors.Addenda.GeneralMotors.Version();
          case 29: return ElectronicDocumentLibrary.Gomsa.Addenda.Gomsa.Version;
          case 30: return ElectronicDocumentLibrary.GrupoModelo.Addenda.GrupoModelo.Version();
          case 31: return ElectronicDocumentLibrary.Heb.Addenda.Heb.Version();
          case 32: return ElectronicDocumentLibrary.Inbursa.Addenda.Inbursa.Version();
          case 33: return ElectronicDocumentLibrary.Itsb.Addenda.ItSmartBusiness.Version();
          case 34: return ElectronicDocumentLibrary.Iusacell.Addenda.Iusacell.Version();
          case 35: return ElectronicDocumentLibrary.Kuehne.Addenda.Kuehne.Version();
          case 36: return ElectronicDocumentLibrary.Lamosa.Addenda.Lamosa.Version();
          case 37: return ElectronicDocumentLibrary.Landsteiner.Addenda.Landsteiner.Version();
          case 38: return ElectronicDocumentLibrary.Loreal.Addenda.Loreal.Version();
          case 39: return ElectronicDocumentLibrary.Lowes.Addenda.Lowes.Version();
          case 40: return ElectronicDocumentLibrary.Mabe.Addenda.Mabe.Version();
          case 41: return ElectronicDocumentLibrary.Maersk.Addenda.Maersk.Version();
          case 42: return ElectronicDocumentLibrary.MultiPack.Addenda.MultiPack.Version();
          case 43: return ElectronicDocumentLibrary.Oxxo.Addenda.Oxxo.Version();
          case 44: return ElectronicDocumentLibrary.Pemex.Addenda.PemexExploracion.Version();
          case 45: return ElectronicDocumentLibrary.Pemex.Addenda.PemexExploracion.Version();
          case 46: return ElectronicDocumentLibrary.Pepsico.Addenda.Pepsico.Version();
          case 47: return ElectronicDocumentLibrary.Pilgrims.Addenda.Pilgrims.Version();
          case 48: return ElectronicDocumentLibrary.Prolec.Addenda.Prolec.Version();
          case 49: return ElectronicDocumentLibrary.Sanmina.Addenda.Sanmina.Version();
          case 50: return ElectronicDocumentLibrary.Santander.Addenda.Santander.Version();
          case 51: return ElectronicDocumentLibrary.SectorPrimario.Addenda.SectorPrimario.Version();
          case 52: return ElectronicDocumentLibrary.SevenEleven.Addenda.SevenEleven.Version();
          case 53: return ElectronicDocumentLibrary.SolerPalau.Addenda.SolerPalau.Version();
          case 54: return CargaRemision.Version();
          case 55: return Servicio.Version();
          case 56: return ElectronicDocumentLibrary.TiendasNeto.Addenda.TiendasNeto.Version();
          case 57: return ElectronicDocumentLibrary.TransportesCastores.Addenda.TransportesCastores.Version();
          case 58: return ElectronicDocumentLibrary.TresM.Addenda.TresM.Version;          
          case 59: return ElectronicDocumentLibrary.TvAzteca.Addenda.TvAzteca.Version();
          case 60: return ElectronicDocumentLibrary.Viscofan.Addenda.Viscofan.Version;
          case 61: return ElectronicDocumentLibrary.Volkswagen.Addenda.Volkswagen.Version();
          case 62: return ElectronicDocumentLibrary.WalMart.Addenda.WalMart.Version;
          case 63: return ElectronicDocumentLibrary.ZfMexico.Addenda.ZfMexico.Version();
        }

        return string.Empty;
      }
    }

    #endregion
  }
}